analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -40,11 +41,17 @@ internal class RoslynSymbolicExecution
         public RoslynSymbolicExecution(ControlFlowGraph cfg, SymbolicCheck[] checks)
         {
             this.cfg = cfg ?? throw new ArgumentNullException(nameof(cfg));
-            this.checks = checks ?? throw new ArgumentNullException(nameof(checks));
-            if (!checks.Any())
+
+            if (checks == null)
+            {
+                throw new ArgumentNullException(nameof(checks));
+            }
+            else if (checks.Length == 0)
```

Comment:
As this is only dev-time safetynet, we can probably simplify it to a single exception:
```suggestion
            if (checks == null || checks.Length == 0)
```
The `At least one check is expected` message looks good enough.